File: app/src/test/java/com/gytxtx/openjbd/PermissionRecoveryTest.java
package com.gytxtx.openjbd;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public final class PermissionRecoveryTest {
    @Test
    public void rationaleAvailableRequestsPermissionAgain() {
        assertFalse(DeviceListActivity.shouldOpenAppSettings(true));
    }

    @Test
    public void rationaleUnavailableOpensAppSettings() {
        assertTrue(DeviceListActivity.shouldOpenAppSettings(false));
    }
}

File: app/src/test/kotlin/com/gytxtx/openjbd/PermissionRecoveryTest.kt
package com.gytxtx.openjbd

import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class PermissionRecoveryTest {
    @Test
    fun rationaleAvailableRequestsPermissionAgain() {
        assertFalse(DeviceListActivity.shouldOpenAppSettings(true))
    }

    @Test
    fun rationaleUnavailableOpensAppSettings() {
        assertTrue(DeviceListActivity.shouldOpenAppSettings(false))
    }
}
